module/platform/webots/Webots/src/Webots.cpp

```diff
@@ -321,29 +342,27 @@ namespace module::platform::webots {
             }
         });
 
-        error_io = on<IO>(fd, IO::CLOSE | IO::ERROR).then([this](const IO::Event& event) {
+        error_io = on<IO>(fd, IO::CLOSE | IO::ERROR).then([this, server_address, port](const IO::Event& /*event*/) {
             // Something went wrong, reopen the connection
             setup_connection(server_address, port);
         });
 
         shutdown_handle = on<Shutdown>().then([this] {
             // Disconnect the fd gracefully
-            shutdown(fd, SHUT_RDWR);
-            close(fd);
-            fd = -1;
+            shutdown(this->fd, SHUT_RDWR);
+            close(this->fd);
+            this->fd = -1;
```

Comment:
No need for `this`

Reply:
there is a need for `this`. the class variable `fd` can't be captured by the lambda. It's either this, or `this->get_fd/set_fd`

Reply:
You don't need the this if you capture the [this] in the lambda